### ISSUE TITLE
False positive trait bound T: Clone is not satisfied

### ISSUE THREAD

**Issue by qiujiangkun** [intellij-rust/intellij-rust#7275]
## Environment

* **IntelliJ Rust plugin version:** 0.4.147.3871-211
* **Rust toolchain version:** 1.54.0-nightly (4e3e6db01 2021-05-18) x86_64-unknown-linux-gnu
* **IDE name and version:** CLion 2021.1.1 (CL-211.7142.21)
* **Operating system:** Linux 5.10.36-2-manjaro
* **Macro expansion engine:** new
* **Name resolution engine:** new

```rust
pub trait Foo: Copy + Clone  {}
impl<T: Copy + Clone> Foo for T {}
```
![image](https://user-images.githubusercontent.com/33482468/120092185-b8a93a80-c143-11eb-8027-c32b2351687a.png)
